File: examples/fsm_platformer/fsm_platformer_python/player_idle_state.py
"""Player Idle State — default state with no movement."""

from __future__ import annotations

from src.engine.core import MonoBehaviour
from src.engine.math.vector import Vector2

from fsm_platformer_python.fsm import FSMState


class PlayerIdleState(FSMState):
    """Player is standing still on the ground."""

    def do_before_entering(self) -> None:
        super().do_before_entering()

    def act(self, owner: MonoBehaviour) -> None:
        from fsm_platformer_python.player_input_handler import PlayerInputHandler
        player: PlayerInputHandler = owner  # type: ignore

        # Zero horizontal velocity, preserve vertical for gravity
        rb = player.rb
        rb.velocity = Vector2(0, rb.velocity.y)

File: examples/fsm_platformer/fsm_platformer_unity/PlayerIdleState.cs
using UnityEngine;

// Unity APIs used: MonoBehaviour, Rigidbody2D, Vector2
public class PlayerIdleState : FSMState
{
    public override void DoBeforeEntering()
    {
        base.DoBeforeEntering();
    }

    public override void Act(MonoBehaviour owner)
    {
        PlayerInputHandler player = (PlayerInputHandler)owner;
        Rigidbody2D rb = player.Rb;
        rb.linearVelocity = new Vector2(0, rb.linearVelocity.y);
    }
}
